### PR TITLE
use dl.k8s.io, not kubernetes-release bucket

### PR DESCRIPTION
**Pull request by BenTheElder**
the linkcheck tool also references this but isn't used anywhere that I Can find, so just taking the opportunity to delete it, other repos use other tools

### CODE REVIEW

**Comment by BenTheElder** [comment]
ref: https://github.com/kubernetes/k8s.io/issues/2396

File: images/kubekins-e2e/Dockerfile [hunk 1]
```diff
@@ -59,8 +59,8 @@ RUN wget -q -O cfssl "https://pkg.cfssl.org/${CFSSL_VERSION}/cfssl_linux-amd64"
 # replace kubectl with one from K8S_RELEASE
 ARG K8S_RELEASE=latest
 RUN rm -f $(which kubectl) && \
-    export KUBECTL_VERSION=$(curl https://storage.googleapis.com/kubernetes-release/release/${K8S_RELEASE}.txt) && \
-    wget https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl -O /usr/local/bin/kubectl && \
+    export KUBECTL_VERSION=$(curl -L https://dl.k8s.io/release/${K8S_RELEASE}.txt) && \
```
Comment (BenTheElder) [inline images/kubekins-e2e/Dockerfile:62]: -L to follow redirect, same for the e2e_node curl

**Comment by BenTheElder** [comment]
filed https://github.com/kubernetes/test-infra/pull/29496 for pull-test-infra-integration, it's not related to this PR at all

**Comment by BenTheElder** [comment]
The test failure doesn't make sense, it's complaining about go.mod ???

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29495/pull-test-infra-verify-lint/1656420211999379456

**Comment by BenTheElder** [comment]
Fixed go mod tidy

File: images/kubekins-e2e/Dockerfile (hunk 1 above)
Comment (dims) [reply images/kubekins-e2e/Dockerfile:62]: good one!

**Comment by dims** [comment]
/approve
/lgtm

/hold for additional 👀 if needed

good catch of the `-L`!

**Pull request merged**